src/trace/listener.ts

```diff
@@ -147,8 +159,12 @@ export class TraceListener {
         ...this.stepFunctionContext,
       };
     }
-
-    if (parentSpanContext !== null) {
+    if (this.inferredSpan !== (null || undefined)) {
```

Comment:
Is there a reason for writing it this way? This basically reduces to:

```
this.inferredSpan !== undefined
```

Perhaps, it should just be:

```suggestion
    if (this.inferredSpan) {
```